### ISSUE TITLE
Possible bug at apps/music/utility/util.php

### ISSUE THREAD

**Issue by Biktorgj** [owncloud/music#775]
Hi,
While scanning my library (about 26 Gb), rescan stops at 0 of 1600 songs, and I get a 500 POST error in the browser.
Looking at the logs I see this line repeated:

**Invalid argument supplied for foreach() at /var/www/nextcloud/apps/music/utility/util.php#86**
That line corresponds to **arrayMultiGet** function which is a small as this:
```
        public static function arrayMultiGet(array $array, $indices) {
                $result = [];
                foreach ($indices as $index) {
                        $result[] = $array[$index];
                }
                return $result;
        }
```

Problem 1: It expects two arrays, but it is enforcing only one. 
Problem 2: The $indices array is being declared as null, breaking the function apart. If I insert an if (isarray($indices)) before the foreach statement the code fails further in the server with the following line:
**array_rand(): Array is empty at /var/www/nextcloud/apps/music/utility/random.php#37**
 
Which responds to this function:
  ```
        public static function pickItems($itemArray, $count) {
                $count = \min($count, \count($itemArray)); // can't return more than all items
                $indices = \array_rand($itemArray, $count);
                if ($count == 1) { // return type is not array when randomizing a single index
                        $indices = [$indices];
                }

                return Util::arrayMultiGet($itemArray, $indices);
        }
```
Still haven't found the root of the problem, since my library is not being scanned at all (I guess I could be missing some schema changes from some updates or something like that), but it would be nice if the app didn't try to randomize empty arrays or process null strings if something goes bad in the library...

**Comment by paulijar** [comment]
Thanks for the report.

1) I just added the type hinting to the `Util` class recently, but I seem to have missed that one parameter `$indices`. I'll add that one, too.

2) The function `Util::arrayMultiGet` is used exclusively to implement getting random songs or albums via the Subsonic API. The error logs you saw, seem to appear whenever the Subsonic API method `getRandomSongs` is called on an empty library. The method still returns an empty result as expected. I'll add one more null check to get rid of the logged error. However, I don't think that this has anything to do with your actual scanning problem. It's just that you now have an empty library, and some Subsonic client is polling for random songs in the background (I have seen a Subsonic client doing that, I *think* it was DSub but can't remember for certain).

**Comment by Biktorgj** [comment]
>However, I don't think that this has anything to do with your actual scanning problem. It's just that you now have an empty library, and some Subsonic client is polling for random songs in the background (I have seen a Subsonic client doing that, I _think_ it was DSub but can't remember for certain).

It seems it didn't, I later saw I was receiving the same error as reported in #774, when forcing a rescan via music app after having changed folders furiously and 'resetting the music collection'

Amazing app, by the way, just started using it yesterday and already in love with it. Thank you for it!

**Comment by paulijar** [comment]
I just released Music v0.15.1 fixing also these two issues.
